lexicon/conversion.py

```diff
@@ -72,6 +86,10 @@ def run(self):
                 transform(orth.text) in words
                 or "special" in lemma.attrib
                 or (orth.text is not None and orth.text.startswith("["))
+                or (
+                    self.check_synt_tok
+                    and transform(lemma.find("./synt/tok").text) in words
```

Comment:
What if there are multiple tokens in the synt sequence, e.g.
```
  <lemma>
    <orth>AppTec</orth>
    <phon>æ p t ɛ k</phon>
    <synt>
        <tok> tech </tok>
        <tok> company </tok>
    </synt>
  </lemma>
```
I think in this case this line will only check the first token and not all of them. Maybe use
```suggestion
                    and all([transform(tok.text) in words for tok in lemma.findall("./synt/tok")])
```
so that the word is kept if all tokens are in the list.

Reply:
Good point. I actually thought that there can only be one <tok> per lemma.

Reply:
Isnt it enough if ONE tok is in the list to have the lemma used in recognition ?

Reply:
eg exchange any with all? How is the calculation of lm probability done if there are multiple tokens in a lemma? (Not that I have seen it IRL though :sunglasses: )

Reply:
> Isnt it enough if ONE tok is in the list to have the lemma used in recognition ?

this depends on what the use case is. In case you want to filter out all words that are not in the LM, then I think all words should be there, because

> How is the calculation of lm probability done if there are multiple tokens in a lemma?

this is the product of the probabilities (or rather sum of log_prob) `p(tok1 | history) *  p(tok2 | history tok1) * ...`

Reply:
my question is if it would crash or become 0? log_prob is probably okay.
And if in this case we would like to remove the tok not on the lm and not the whole lemma, which would be happening now.

Reply:
> my question is if it would crash or become 0? log_prob is probably okay.

if e.g. tok2 didn't exist in the LM, then `p(tok2 | history tok1)` would be replaced with `p(<unk>| history tok1)`. If there is no unknown in LM, then the probability will probably be 0, i.e. log_prob = some_large_float. 

>  if in this case we would like to remove the tok not on the lm and not the whole lemma,

I don't have a strong argument here.. my gut feeling would be to remove it entirely and not to modify the lemma..

Reply:
Yes, i think I agree. Lets do that. As I said: I Have never seen a lemma with multiple tokens ever. :-)